### ISSUE TITLE
Inconsistency in storage interfaces

### ISSUE THREAD

**Issue by afilippov1985** [bshaffer/oauth2-server-php#675]
Somewhere function must return `false` if value not exist (`ClientInterface::getClientDetails()`)
Somewhere function must return `null` in such case (`AuthorizationCodeInterface::getAuthorizationCode()`)
Somewhere behavior not defined (`PublicKeyInterface::getPrivateKey()`)

I suggest to revise interfaces and
return `null` if value not exist (not defined)
return `false` on error

**Comment by bshaffer** [comment]
This seems like a good idea. I seem to remember returning `false` for a specific reason, but it may have just been because `Pdo` returns false from its queries, which is not a real good reason.

**Comment by afilippov1985** [comment]
Also, existing storages has functions to set data for unit tests (such as `setClientDetails`, `setScope`, `setClientKey`, `unsetAccessToken`, `setUser`). They should be added to corresponding storage interfaces.

**Comment by bshaffer** [comment]
I disagree with this second request. These are convenience functions, and if you add your own storage classes, you are free to add those functions or not add those functions.  The library classes require them for testing, but the actual OAuth server does not require them.
